examples/basics/gloo/spatial_filters.py

```diff
@@ -0,0 +1,163 @@
+    }
+    else if (u_interpolation < 5.5) {
+        gl_FragColor = Kaiser(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 6.5) {
+        gl_FragColor = Quadric(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 7.5) {
+        gl_FragColor = Bicubic(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 8.5) {
+        gl_FragColor = CatRom(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 9.5) {
+        gl_FragColor = Mitchell(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 10.5) {
+        gl_FragColor = Spline16(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 11.5) {
+        gl_FragColor = Spline36(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 12.5) {
+        gl_FragColor = Gaussian(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 13.5) {
+        gl_FragColor = Bessel(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 14.5) {
+        gl_FragColor = Sinc(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 15.5) {
+        gl_FragColor = Lanczos(u_texture, u_shape, v_texcoord);
+    }
+    else gl_FragColor = Blackman(u_texture, u_shape, v_texcoord);
+}
+
+"""
+
+
+class Canvas(app.Canvas):
+
+    def __init__(self):
+        app.Canvas.__init__(self, keys='interactive', size=((512), (512)))
+
+        self.program = gloo.Program(VERT_SHADER, FRAG_SHADER)
+        self.texture = gloo.Texture2D(I, interpolation='nearest')
+
+        # using internalformat 'r16f' as discussed in issue #1017
+        self.kernel = gloo.Texture2D(kernel, interpolation='nearest',
+                                     internalformat='r16f')
+
+        self.program['u_texture'] = self.texture
+        self.program['u_kernel'] = self.kernel
+        self.program['u_shape'] = I.shape[1], I.shape[0]
+        self.program['u_interpolation'] = 0
+
+        self.names = [
+            "Nearest", "Bilinear", "Hanning", "Hamming",
+            "Hermite", "Kaiser", "Quadric", "Bicubic",
+            "CatRom", "Mitchell", "Spline16", "Spline36",
+            "Gaussian", "Bessel", "Sinc", "Lanczos", "Blackman"]
+
+        self.title = 'Spatial Filtering using %s Filter' % \
+                     self.names[int(self.program['u_interpolation'])]
+
+        self.program.bind(gloo.VertexBuffer(data))
+
+        gloo.set_clear_color('white')
+
+        self.show()
+
+    def on_key_press(self, event):
+        if event.key in ['Left', 'Right']:
+            if event.key == 'Right':
+                step = 1
+            else:
+                step = -1
+            self.program['u_interpolation'] = \
+                (self.program['u_interpolation'] + step) % 17
+        self.title = 'Spatial Filtering using %s Filter' % \
+                     self.names[int(self.program['u_interpolation'])]
+        self.update()
```

Comment:
these can be indented as well, as you only need an update if the interp has changed